pkg/assisted/agentserviceconfig_test.go

```diff
@@ -0,0 +1,810 @@
+		}
+
+		testBuilder := NewAgentServiceConfigBuilder(
+			testSettings, testcase.databaseStorage, testcase.filesystemStorage)
+
+		if testcase.client {
+			assert.Equal(t, testcase.databaseStorageSize,
+				*testBuilder.Definition.Spec.DatabaseStorage.Resources.Requests.Storage())
+			assert.Equal(t, testcase.filesystemStorageSize,
+				*testBuilder.Definition.Spec.FileSystemStorage.Resources.Requests.Storage())
+			assert.Empty(t, testBuilder.errorMsg)
+		} else {
+			assert.Nil(t, testBuilder)
+		}
+	}
+}
+
+func TestNewDefaultAgentServiceConfigBuilder(t *testing.T) {
+	testCases := []struct {
+		client            bool
+		databaseStorage   resource.Quantity
+		filesystemStorage resource.Quantity
+	}{
+		{
+			client:            true,
+			databaseStorage:   resource.MustParse(defaultDatabaseStorageSize),
+			filesystemStorage: resource.MustParse(defaultFilesystemStorageSize),
+		},
+		{
+			client:            false,
+			databaseStorage:   resource.MustParse(defaultDatabaseStorageSize),
+			filesystemStorage: resource.MustParse(defaultFilesystemStorageSize),
+		},
+	}
+
+	for _, testcase := range testCases {
+		var testSettings *clients.Settings
+		if testcase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{})
+		}
+
+		testBuilder := NewDefaultAgentServiceConfigBuilder(testSettings)
+
+		if testcase.client {
+			assert.Equal(t, testcase.databaseStorage,
+				*testBuilder.Definition.Spec.DatabaseStorage.Resources.Requests.Storage())
+			assert.Equal(t, testcase.filesystemStorage,
+				*testBuilder.Definition.Spec.FileSystemStorage.Resources.Requests.Storage())
+			assert.Empty(t, testBuilder.errorMsg)
+		} else {
+			assert.Nil(t, testBuilder)
+		}
+	}
+}
+
+func TestPullAgentServiceConfig(t *testing.T) {
+	testCases := []struct {
+		client        bool
+		exists        bool
+		expectedError error
+	}{
+		{
+			client:        true,
+			exists:        true,
+			expectedError: nil,
+		},
+		{
+			client:        false,
+			exists:        true,
+			expectedError: errors.New("the apiClient is nil"),
+		},
+		{
+			client:        true,
+			exists:        false,
+			expectedError: errors.New("agentserviceconfig object agent does not exist"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateAgentServiceConfig())
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{K8sMockObjects: runtimeObjects})
+		}
+
+		testBuilder, err := PullAgentServiceConfig(testSettings)
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError != nil {
+			assert.Nil(t, testBuilder)
+		} else {
+			assert.NotNil(t, testBuilder)
+			assert.Equal(t, agentServiceConfigName, testBuilder.Object.Name)
+			assert.Empty(t, testBuilder.errorMsg)
+		}
+	}
+}
+
+func TestAgentServiceConfigGet(t *testing.T) {
```

Comment:
The error scenario runs for `exists: false` test. In this case I am just checking that err != nil because the error is coming from an external pkg that we do not control

Reply:
```
&errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"agentserviceconfigs.agent-install.openshift.io \"agent\" not found", Reason:"NotFound", Details:(*v1.StatusDetails)(0xc000659500), Code:404}}
```

Reply:
Yes that's ok. You can just compare strings by calling > err.Error()